[Integ-tests] Improve integration tests ARNs to be compatible with other AWS partitions

hanwen-pcluste opened the pull request:
### Description of changes
See commit descriptions for details

### Tests
The following integration tests have been passed
```
{%- import 'common.jinja2' as common -%}
{%- set NEW_REGION = ["us-east-1"] -%}
---
test-suites:
  cli_commands:
    test_cli_commands.py::test_slurm_cli_commands:
      dimensions:
        - regions: ["ap-northeast-2"]
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: ["alinux2"]
          schedulers: ["slurm"]
  iam:
    test_iam.py::test_iam_policies:
      dimensions:
        - regions: {{ NEW_REGION }}
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: ["alinux2"]
          schedulers: ["slurm"]
    test_iam.py::test_iam_resource_prefix:
      dimensions:
        - regions: [ "eu-north-1" ]
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: [ "alinux2" ]
          schedulers: [ "slurm" ]
  schedulers:
    test_slurm.py::test_slurm_config_update:
      dimensions:
        - regions: ["ap-east-1"]
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: ["alinux2"]
          schedulers: ["slurm"]
    test_slurm.py::test_slurm_memory_based_scheduling:
      dimensions:
        - regions: ["ap-east-1"]
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: ["alinux2"]
          schedulers: ["slurm"]
  storage:
    test_ephemeral.py::test_head_node_stop:
      dimensions:
        - regions: ["us-east-1"]
          instances: ["m5d.xlarge", "h1.2xlarge"]
          oss: ["alinux2"]
          schedulers: ["slurm"]
    test_efs.py::test_efs_compute_az:
      dimensions:
        - regions: ["eu-west-2"]
          instances: {{ common.INSTANCES_DEFAULT_X86 }}
          oss: ["alinux2"]
          schedulers: ["slurm"]
```

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

hanwen-cluster committed:
[Integ-tests] Adjust assert_subnet_az_relations_from_config for isolated regions

Some isolated regions do not have three availability zones. This commit relaxes the check for isolated regions.

Signed-off-by: Hanwen <hanwenli@amazon.com>

gmarciani commented on tests/integration-tests/tests/cli_commands/test_cli_commands.py:
```diff
@@ -310,13 +312,13 @@ def _test_pcluster_export_cluster_logs(s3_bucket_factory, cluster):
                 "Action": "s3:PutObject",
                 "Effect": "Allow",
-                "Resource": f"arn:aws:s3:::{bucket_name}/*",
+                "Resource": f"arn:{partition}:s3:::{bucket_name}/*",
```
What about retrieving the partition from the cluster rather than adding a new parameter?
It would be better to have something like:

```
cluster.partition
```

or

```
get_partition(cluster.region)
```

I think it would be a better approach because it's highly probable that whenever we need an ARN, we have the cluster or cluster region information already available in the scope (like in this specific case).

hanwen-pcluste replied:
Done

gmarciani commented on tests/integration-tests/tests/schedulers/test_slurm/test_slurm_config_update/pcluster.config.update.yaml:
```diff
@@ -8,7 +8,7 @@ HeadNode:
   Iam:
     AdditionalIamPolicies:
-      - Policy: arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+      - Policy: arn:{{partition}}:iam::aws:policy/AmazonSSMManagedInstanceCore
```
We should parameterize the partition in this way in many other yaml file I suppose.
For example: I see a policy ARN `arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess` also in `/Users/mgiacomo/workplace/aws-parallelcluster-dev/aws-parallelcluster/cli/tests/pcluster3_config_converter/test_pcluster3_config_converter/test_pcluster3_config_converter/slurm_full.yaml`.

Let's fix the ARN issue everywhere in integ tests

hanwen-pcluste replied:
The ones you mentioned are from unit tests. We will keep them unchanged because this PR focuses on integration tests

gmarciani replied:
Ok

gmarciani commented on tests/integration-tests/tests/schedulers/test_slurm.py:
```diff
@@ -1890,9 +1902,9 @@ def _test_enable_fast_capacity_failover(
+def _test_update_without_update_queue_params(partition, pcluster_config_reader, cluster, remote_command_executor):
     """Test update without queue param change, clustermgtd and slurmctld not restart."""
-    updated_config_file = pcluster_config_reader(config_file="pcluster.config.update.yaml")
+    updated_config_file = pcluster_config_reader(config_file="pcluster.config.update.yaml", partition=partition)
```
Instead of adding an additional parameter to every method that manipulates the cluster config file, let's create a fixture like the one used for region, that injects the `partition` parameter in every cluster config file

hanwen-pcluste replied:
Done

hanwen-cluster committed:
[Integ-tests] Improve ARNs in integration tests to be compatible with other AWS partitions

1. Add a default variable (AWS partition) to cluster configuration file Jinja template.
2. Add partition variable in Cluster object
3. AD test, Slurm accounting, and tranium tests are not changed. Because we do not need to let them run in other partitions for now

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-pcluste committed:
Merge branch 'release-3.5' into release-3.5mar2
